generator.py

```diff
@@ -368,106 +129,51 @@ def generate_new_sample(template, htmlgrammar, cssgrammar, jsgrammar):
 
     return result
 
-
-def generate_samples(grammar_dir, outfiles):
+def generate_sample(template, outfiles=None):
```

Comment:
This function either returns the generated content or returns nothing and outputs to a file. I don't think that's very clean. If the caller wants to generate a single file they can just pass that single file in the `outfiles` array. No need to implement a separate behavior for generating a single file.